Classes/TYPO3/LDAP/Security/Authentication/Provider/LDAPProvider.php

```diff
@@ -173,6 +215,63 @@ protected function updateParty(Account $account, array $ldapSearchResult) {
 	 * @return void
 	 */
 	protected function setRoles(Account $account, array $ldapSearchResult) {
+		if (is_array($this->rolesConfiguration)) {
+			$contextVariables = array(
+				'ldapUser' => $ldapSearchResult,
+			);
+			if (isset($this->defaultContext) && is_array($this->defaultContext)) {
+				foreach ($this->defaultContext as $contextVariable => $objectName) {
+					$object = $this->objectManager->get($objectName);
+					$contextVariables[$contextVariable] = $object;
+				}
+			}
+			$eelContext = new Context($contextVariables);
+
+			foreach ($this->rolesConfiguration['default'] as $roleIdentifier) {
+				$role = $this->policyService->getRole($roleIdentifier);
+				$account->addRole($role);
+			}
+
+			$dn = $this->eelEvaluator->evaluate($this->partyConfiguration['dn'], $eelContext);
+
```

Comment:
if you're on fixing that other comment about the empty array in the config file I'd remove this empty line (236) too